pkg/uprobetracer/bpf/usdt_helper.bpf.c

```diff
@@ -0,0 +1,363 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+// Copyright (c) 2024 The Inspektor Gadget authors
+
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#define USDT_ARG_VERSION 1
+
+enum USDT_ARG_TYPE {
+	/* Argument in register */
+	USDT_ARG_TYPE_REG,
+	/* Argument in memory */
+	USDT_ARG_TYPE_MEM,
+};
+
+enum USDT_ARG_LENGTH {
+	USDT_ARG_LENGTH_UINT8 = 0,
+	USDT_ARG_LENGTH_INT8,
+	USDT_ARG_LENGTH_UINT16,
+	USDT_ARG_LENGTH_INT16,
+	USDT_ARG_LENGTH_UINT32,
+	USDT_ARG_LENGTH_INT32,
+	USDT_ARG_LENGTH_UINT64,
+	USDT_ARG_LENGTH_INT64,
+};
+
+// using inline asm to avoid the "dereference of modified ctx ptr" error,
+// see https://mejedi.dev/posts/ebpf-dereference-of-modified-ctx-ptr-disallowed/
+#define RETURN_CTX_FIELD(field)                                               \
+	asm("%[res] = *(u64 *)(%[base] + %[offset])"                          \
```

Comment:
Seems not being needed. This asm is used for preventing the compiler from using some "table-driven" optimizations (I even tried compiling with `-fno-jump-tables` but it's still using table-driven approaches), which will cause the object failing to pass the verification.

We don't care if the compiler wants to merge some load instructions, it's good as long as it could make the verifier happy.